buildscripts/checkdeps.sh

```diff
@@ -25,6 +25,11 @@ _init() {
     OSX_VERSION="10.8"
     KNAME=$(uname -s)
     ARCH=$(uname -m)
```

Comment:
On Solaris, uname -m returns the type of processor. However, Solaris is dual 32/64-bit so all that uname -m tells you is that you're running on Intel (as opposed to sparc or whatever) - it doesn't tell you whether you're 32 or 64-bit. The isainfo command tells you which instruction set architectures are available; isainfo -k tells you which variant the kernel is running in.

See below for the output on my system:

% uname -m
i86pc
% isainfo
amd64 i386
% isainfo -k
amd64

The checkdeps.sh script is looking for what isainfo -k gives on a Solaris system.